heatmapchart/src/components/HeatMapChartPanel.tsx

```diff
@@ -0,0 +1,191 @@
+      return {
+        data: [],
+        xAxisCategories: [],
+        yAxisCategories: [],
+        countMin: 0,
+        countMax: 0,
+        timeScale: undefined,
+      };
+    }
+
+    if (
+      queryResults.length != 1 ||
+      queryResults[0]!.data.series.length != 1 ||
+      queryResults[0]!.data.series[0]!.histograms === undefined
+    ) {
+      return {
+        data: [],
+        xAxisCategories: [],
+        yAxisCategories: [],
+        countMin: 0,
+        countMax: 0,
+        timeScale: undefined,
+      };
+    }
+
+    const series: TimeSeries = queryResults[0]!.data.series[0]!;
+
+    const timeScale = getCommonTimeScaleForQueries(queryResults);
+    const xAxisCategories: number[] = generateCompleteTimestamps(timeScale);
+
+    // Dummy value that will be replaced at the first iteration
+    let lowestBound = Infinity;
+    let highestBound = -Infinity;
+    let countMin = Infinity;
+    let countMax = -Infinity;
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const [_, histogram] of series?.histograms ?? []) {
+      for (const bucket of histogram?.buckets ?? []) {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const [_, lowerBound, upperBound, count] = bucket;
+        const lowerBoundFloat = parseFloat(lowerBound);
+        const upperBoundFloat = parseFloat(upperBound);
+        const countFloat = parseFloat(count);
+        if (lowerBoundFloat < lowestBound) {
+          lowestBound = lowerBoundFloat;
+        }
+        if (upperBoundFloat > highestBound) {
+          highestBound = upperBoundFloat;
+        }
+        if (countFloat < countMin) {
+          countMin = countFloat;
+        }
+        if (countFloat > countMax) {
+          countMax = countFloat;
+        }
+      }
+    }
+
+    const height = contentDimensions?.height ?? HEATMAP_MIN_HEIGHT;
+    const totalRange = highestBound - lowestBound;
+    const rangePerItem = (totalRange * HEATMAP_ITEM_MIN_HEIGHT) / height;
+    const totalItems = Math.ceil(height / HEATMAP_ITEM_MIN_HEIGHT);
+
+    // Generating value of the Y axis based on the height divided by the size of a cell (item)
+    const yAxisCategories: string[] = Array.from({ length: totalItems }, (_, index) =>
+      (lowestBound + index * rangePerItem).toFixed(3)
+    );
+
+    const data: HeatMapDataItem[] = [];
+    // Logic for filling all cells where a bucket is present
+    for (const [time, histogram] of series?.histograms ?? []) {
+      const itemIndexOnXaxis = xAxisCategories.findIndex((v) => v === time * 1000);
+
+      for (const bucket of histogram?.buckets ?? []) {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const [_, lowerBound, upperBound, count] = bucket;
+        const yLowerBoundItem = Math.floor((parseFloat(lowerBound) - lowestBound) / rangePerItem);
+        const yUpperBoundItem = Math.ceil((parseFloat(upperBound) - lowestBound) / rangePerItem);
```

Comment:
I'm not sure but shouldnt it be
```suggestion
        const yUpperBoundItem = Math.ceil((parseFloat(upperBound) - highestBound) / rangePerItem);
```
here?

Reply:
Nope, it's for putting lower bound item at index 0 and upper item need to be reduce by the same amount